validator/client/validator.go

```diff
@@ -1224,6 +1224,10 @@ func (v *validator) filterAndCacheActiveKeys(ctx context.Context, pubkeys [][fie
 
 // updateValidatorStatusCache updates the validator statuses cache, a map of keys currently used by the validator client
 func (v *validator) updateValidatorStatusCache(ctx context.Context, pubkeys [][fieldparams.BLSPubkeyLength]byte) error {
+	if len(pubkeys) == 0 {
+		v.pubkeyToStatus = make(map[[fieldparams.BLSPubkeyLength]byte]*validatorStatus, 0)
```

Comment:
This looks like it could overwrite an existing cache? I'm not sure why we would need to do this here.

Reply:
It's the validators statuses cache, it's used to track what validators are available and their statuses on the validator client. In this case if it's updated to nil or no keys we should override it. Typically this is handled as the api endpoint would return no statuses and it would be overridden below but the validator rest implementation of this didn't handle 0 keys correctly.